### PR TITLE
Data visitor

### PR DESCRIPTION
**Pull request by Brezelfelder**
data visitor and Logic vor Language Validation in Prismic Documents

### CODE REVIEW

File: src/Structure/Field/EmbedField.php [hunk 1]
```diff
@@ -91,9 +92,18 @@ public function validateData (DataValidator $validator, array $path, mixed $data
...
+	 *
+	 * @template T
+	 * @param T $data
+	 * @return T
 	 */
-	public function transformValue (mixed $data, DataTransformer $dataTransformer) : VideoValue
+	public function transformValue (
+		mixed $data,
+		DataTransformer $dataTransformer,
+		?DataVisitorInterface  $dataVisitor = null,
```
Comment (apfelbox) [inline src/Structure/Field/EmbedField.php]:
```suggestion
		?DataVisitorInterface $dataVisitor = null,
```

File: src/Structure/Field/GeoPointField.php [hunk 2]
```diff
@@ -55,9 +56,19 @@ public function validateData (DataValidator $validator, array $path, mixed $data
...
+	 *
+	 * @template T
+	 * @param T $data
+	 * @return T
 	 */
-	public function transformValue (mixed $data, DataTransformer $dataTransformer) : ?array
+	public function transformValue (
+		mixed $data,
+		DataTransformer $dataTransformer,
+		?DataVisitorInterface  $dataVisitor = null,
```
Comment (apfelbox) [inline src/Structure/Field/GeoPointField.php]:
```suggestion
		?DataVisitorInterface $dataVisitor = null,
```

File: src/Structure/Field/GroupField.php [hunk 3]
```diff
@@ -63,9 +64,19 @@ public function validateData (DataValidator $validator, array $path, mixed $data
...
+	 *
+	 * @template T
+	 * @param T $data
+	 * @return T
 	 */
-	public function transformValue (mixed $data, DataTransformer $dataTransformer) : array
+	public function transformValue (
+		mixed $data,
+		DataTransformer $dataTransformer,
+		?DataVisitorInterface  $dataVisitor = null,
```
Comment (apfelbox) [inline src/Structure/Field/GroupField.php]:
```suggestion
		?DataVisitorInterface $dataVisitor = null,
```

File: src/Structure/Field/IntegrationField.php [hunk 4]
```diff
@@ -1,9 +1,11 @@
-<?php declare(strict_types=1);
+src/Structure/Field/IntegrationField.php<?php declare(strict_types=1);
```
Comment (apfelbox) [inline src/Structure/Field/IntegrationField.php]: Unintended change?

File: src/Structure/Field/IntegrationField.php [hunk 5]
```diff
@@ -35,8 +37,30 @@ public function __construct (
...
+		mixed $data,
+		DataTransformer $dataTransformer,
+		?DataVisitorInterface $dataVisitor = null,
+	) : mixed
+	{
+		$dataVisitor?->onDataVisit($this, $data);
+
+		return $data;
+	}
+}
```
Comment (apfelbox) [inline src/Structure/Field/IntegrationField.php]: Please configure your IDE to always add a line break at the end of the file

**Comment by Brezelfelder** [comment]
@apfelbox  

Anpassungen sind gemacht.

File: src/Visitor/Translation/TranslationCheckVisitor.php [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+
+	/**
+	 */
+	public function onDataVisit(
+		PrismicTypeInterface $field,
+		mixed $data,
+	) : void
+	{
+
```
Comment (apfelbox) [inline src/Visitor/Translation/TranslationCheckVisitor.php]:
```suggestion
```

File: src/Visitor/Translation/TranslationCheckVisitor.php [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+							{
+								$this->issue[] = [
+									"severity" => "ERROR",
+									"message" => "RTE - Wrong locale in Link.",
+									"field" => $field,
+								];
+							}
+						}
+
+						if (!empty($span["type"]))
```
Comment (apfelbox) [inline src/Visitor/Translation/TranslationCheckVisitor.php]: We should only add it once per RTE field. Right now we would add it per paragraph

**Comment by Brezelfelder** [comment]
Part 2 done.

**Pull request merged**